app/service/data_svc.py

```diff
@@ -159,3 +159,9 @@ def __init__(self, dao, utility_svc):
         group = await self.dao.get('core_group', dict(id=group_id))
         await self.dao.update(table='core_group', key='id', value=group_id, data=dict(deactivated=datetime.now().strftime('%Y-%m-%d %H:%M:%S')))
         return 'Removed %s host group' % group[0]['name']
+    
+    async def delete_agent(self, id):
+        agent = await self.dao.get('core_agent', dict(id=id))
```

Comment:
i don't think we need to get the agent here; we're only using the ID, which is passed in, which we should have delete ability on. i could see an argument for doing it for validity-sake, but i don't see much of a value here-- if anything, i would have the delete always check for validity before allowing a delete (but again, don't think it's worth it)